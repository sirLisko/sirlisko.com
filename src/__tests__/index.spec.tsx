import React from "react";
import { render } from "@testing-library/react";

import Index from "../pages/index";

jest.mock("../../data/me.ts", () => ({
  descriptions: ["foo", "bar"],
  keywords: ["k_foo", "k_bar"],
  links: [
    { name: "foo", label: "Foo", url: "https://foo" },
    { name: "bar", label: "Bar", url: "https://Bar" },
  ],
}));

describe("Index Page", () => {
  it("should render properly", () => {
    const { asFragment } = render(<Index />);
    expect(asFragment()).toMatchSnapshot();
  });
});
