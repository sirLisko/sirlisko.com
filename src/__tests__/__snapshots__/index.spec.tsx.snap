// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Page should render properly 1`] = `
<DocumentFragment>
  <div
    class="home"
  >
    ;
    <section
      class="life"
    >
      <span
        class="heart"
      />
      <span
        class="heart"
      />
      <span
        class="heart"
      />
    </section>
    <figure
      class="ghost"
      style="left: -200px; top: -300px;"
    />
    <header>
      <ul
        class="links"
      >
        <li
          class="foo"
        >
          <a
            href="https://foo"
            title="foo"
          >
            <span
              class="caption"
            >
              Foo
            </span>
          </a>
        </li>
        <li
          class="bar"
        >
          <a
            href="https://Bar"
            title="bar"
          >
            <span
              class="caption"
            >
              Bar
            </span>
          </a>
        </li>
      </ul>
    </header>
    <article>
      <div
        class="container"
      >
        <img
          alt="sirlisko face"
          class="face"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height="714"
          src="./images/sirlisko.svg"
          style="color: transparent;"
          width="496"
        />
        <div
          class="intro"
        >
          <p>
            Ciao!
          </p>
          <p>
            I’m sirLisko,
          </p>
          <p>
            another f@$#&n’
          </p>
          <p
            class="desc"
          >
            <span
              class="descActive"
            >
              foo
            </span>
            <span>
              bar
            </span>
          </p>
        </div>
      </div>
    </article>
    <footer>
      <a
        class="my__projects"
        href="/projects"
      >
        Projects
      </a>
      <a
        class="my__resume"
        href="/resume"
      >
        Resume
      </a>
    </footer>
  </div>
</DocumentFragment>
`;
